commons-packet/commons-packet-manager/src/main/java/io/mosip/commons/packet/util/PacketValidator.java

```diff
@@ -32,294 +38,332 @@
+			List<String> hashValues = fieldValueArray.getValue();
+			hashValues.forEach(value -> {
+				byte[] valuebyte = null;
+				try {
+					InputStream fileStream = checksumMap.get(value);
+					valuebyte = IOUtils.toByteArray(fileStream);
+					outputStream.write(valuebyte);
+				} catch (IOException e) {
+					LOGGER.error("Exception while generating hash " + ExceptionUtils.getStackTrace(e));
+				}
+			});
+		}
+
+		return HMACUtils2.digestAsPlainText(outputStream.toByteArray()).getBytes();
+
+	}
+
+	private Map<String, String> getMetaInfoJson(Packet packet) throws PacketKeeperException, IOException {
+		Map<String, String> finalMap = new HashMap<>();
+		InputStream metaInfoJson = ZipUtils.unzipAndGetFile(packet.getPacket(), "PACKET_META_INFO");
+		if (metaInfoJson != null) {
+			byte[] bytearray = IOUtils.toByteArray(metaInfoJson);
+			String jsonString = new String(bytearray);
+			LinkedHashMap<String, Object> currentIdMap = (LinkedHashMap<String, Object>) mapper
+					.readValue(jsonString, LinkedHashMap.class).get(IDENTITY);
+
+			currentIdMap.keySet().stream().forEach(key -> {
+				try {
+					finalMap.putIfAbsent(key,
+							currentIdMap.get(key) != null ? JsonUtils.javaObjectToJsonString(currentIdMap.get(key))
+									: null);
+				} catch (io.mosip.kernel.core.util.exception.JsonProcessingException e) {
+					throw new GetAllMetaInfoException(e.getMessage());
+				}
+			});
+		}
+		return finalMap;
+	}
+
+	private boolean validateFiles(List hashseq1List, List hashseq2List, Map<String, InputStream> checksumMap,
+			Packet packet) throws JsonProcessingException, IOException {
+		List<String> allFileNames = new ArrayList<>();
+		if (hashseq1List != null && !hashseq1List.isEmpty()) {
+			for (Object o : hashseq1List) {
+				FieldValueArray fieldValueArray = mapper.readValue(JsonUtils.javaObjectToJsonString(o),
+						FieldValueArray.class);
+				allFileNames.addAll(fieldValueArray.getValue());
+			}
+		}
+
+		if (hashseq2List != null && !hashseq2List.isEmpty()) {
+			for (Object o : hashseq2List) {
+				FieldValueArray fieldValueArray = mapper.readValue(JsonUtils.javaObjectToJsonString(o),
+						FieldValueArray.class);
+				allFileNames.addAll(fieldValueArray.getValue());
+			}
+		}
+
+		// added this two file which are extracted checksum
+		allFileNames.add("PACKET_DATA_HASH");
+		allFileNames.add("PACKET_OPERATIONS_HASH");
+
+		/*
+		 * for (String fileName : allFileNames) { InputStream inputStream =
+		 * ZipUtils.unzipAndGetFile(packet.getPacket(), fileName); if (inputStream !=
+		 * null && inputStream.available() > 0) checksumMap.put(fileName, inputStream);
+		 * notFoundFiles.remove(fileName); }
+		 */
+
+		allFileNames = allFileNames.parallelStream().map(x -> FilenameUtils.normalize(x.toLowerCase()))
+				.collect(Collectors.toList());
+
+		List<String> notFoundFiles = new ArrayList<>();
+		allFileNames.forEach(v -> notFoundFiles.add(v));
+		ZipUtils.extracted(checksumMap, packet, allFileNames, notFoundFiles);
```

Comment:
I think this method name should be changed.

Reply:
sorry file was not added by mistake

Reply:
Resolved